protobufs/livekit_agent.proto

```diff
@@ -21,154 +21,174 @@ option ruby_package = "LiveKit::Proto";
 
 import "livekit_models.proto";
 
+service AgentService {
+  rpc StartAgentJob(StartAgentJobRequest) returns (Job);
+  rpc StopAgentJob(StopAgentJobRequest) returns (Job); 
+}
+
+message StartAgentJobRequest {
+  JobType type = 1;
+  Room room = 2;
+  optional ParticipantInfo participant = 3;
+  string namespace = 4;
+  map<string, string> metadata = 5;
```

Comment:
Can this just be a string?

Reply:
We currently use a map for egress, a string for other entities (participant, etc). Either has pros and cons. The map lets us potentially stash fields in there as well using a reserved namespace. The string is more flexible...

Reply:
in cases where we might be tempted to extend the user supplied metadata with values from the service we should probably add structured fields. for example we might include an optional field for sip data with a struct containing phone #, trunk id, etc...